ci/authn-azure/check_dependencies.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -eo pipefail
+
+check_env_var() {
+  if [[ -z "${!1+x}" ]]; then
+# where ${var+x} is a parameter expansion which evaluates to nothing if var is unset, and substitutes the string x otherwise.
+# https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash/13864829#13864829
+    echo "You must set $1 before running these scripts."
+    exit 1
+  fi
+}
+
+# These variables should come from Conjur via summon
+check_env_var "AZURE_TENANT_ID"
+check_env_var "AZURE_SUBSCRIPTION_ID"
+check_env_var "AZURE_RESOURCE_GROUP"
+check_env_var "AZURE_AUTHN_INSTANCE_USERNAME"
+check_env_var "AZURE_AUTHN_INSTANCE_PASSWORD"
+check_env_var "USER_ASSIGNED_IDENTITY"
+
+# These variables should come from Jenkins
+check_env_var "AZURE_AUTHN_INSTANCE_IP"
+#check_env_var "SYSTEM_ASSIGNED_IDENTITY" # TODO: add this once available
```

Comment:
why is `SYSTEM_ASSIGNED_IDENTITY` coming from Jenkins and not also summon?
Also do we want to set both of these in env at the same time due to incompatibility/invalid combination?

Reply:
> why is SYSTEM_ASSIGNED_IDENTITY coming from Jenkins and not also summon?

Because it is created in each build as each pipeline has its own Azure VM. Same as with the IP.

> do we want to set both of these in env at the same time due to incompatibility/invalid combination?

We will have tests for both use-cases in the same pipeline, then they are both needed.